lib/auth/saml.go

```diff
@@ -158,11 +188,36 @@ func (a *Server) calculateSAMLUser(connector types.SAMLConnector, assertionInfo
 
 	p.traits = services.SAMLAssertionsToTraits(assertionInfo)
 
+	diagInfo := func(infoType types.SSOInfoType, value interface{}) {
+		entry, err := types.NewSSODiagnosticInfo(infoType, value)
+		if err != nil {
+			log.WithError(err).Warn("Failed to serialize SSO diag info.")
+		}
+
+		err = a.Identity.CreateSSODiagnosticInfo(context.Background(), types.KindSAML, request.ID, *entry)
```

Comment:
The context here should either be a TODO, or the function should explicitly take a `context.Context` as the first parameter
```suggestion
		err = a.Identity.CreateSSODiagnosticInfo(context.TODO(), types.KindSAML, request.ID, *entry)
```

Reply:
I've modified the calls so that all callers are passing existing contexts.